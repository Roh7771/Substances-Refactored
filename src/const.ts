export const SUBSTANCE_TO_SHOW_AMOUNT = 20;

export const EMPTY_FIELD_VALUE = '-';

export enum ErrorStatus {
  OK,
  DUPLICATE_CAS_NUMBER,
  LOADING_FAILED,
  WRONG_LOGIN_DATA
}

export enum AuthorizationStatus {
  AUTH,
  NO_AUTH
}

export enum ModalWindowStatus {
  EDIT = 'EDIT',
  CREATE = 'CREATE',
  DELETE = 'DELETE',
  NONE = 'NONE'
}
