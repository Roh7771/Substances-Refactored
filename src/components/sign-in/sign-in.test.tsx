import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SignIn from './sign-in';
import { ErrorStatus } from '../../const';

const noop = () => {};

it('Sign in should render correctly', () => {
  const tree = renderer.create(
    <SignIn
      dispatch={noop}
      isRequestLoading={false}
      errorStatus={ErrorStatus.OK}
      csrfToken="token"
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('Sign in field should be disabled while request loading', () => {
  const tree = renderer.create(
    <SignIn
      dispatch={noop}
      isRequestLoading
      errorStatus={ErrorStatus.OK}
      csrfToken="token"
    />,
  );
  expect(tree).toMatchSnapshot();
});

it('Sign in should show an error', () => {
  const tree = renderer.create(
    <SignIn
      dispatch={noop}
      isRequestLoading={false}
      errorStatus={ErrorStatus.WRONG_LOGIN_DATA}
      csrfToken="token"
    />,
  );
  expect(tree).toMatchSnapshot();
});