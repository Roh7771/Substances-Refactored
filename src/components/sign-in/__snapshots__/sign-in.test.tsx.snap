// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in field should be disabled while request loading 1`] = `
<form
  className="signin-form"
  onSubmit={[Function]}
>
  <h1
    className="mb-3 text-center"
  >
    Sign In
  </h1>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      aria-describedby="emailHelp"
      className="form-control"
      disabled={true}
      id="name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="form-control"
      disabled={true}
      id="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="alert alert-danger"
    role="alert"
    style={
      Object {
        "display": "none",
      }
    }
  >
    Неверный логин или пароль.
  </div>
  <button
    className="btn btn-primary signin-form__button"
    disabled={true}
    type="submit"
  >
    Log In
  </button>
</form>
`;

exports[`Sign in should render correctly 1`] = `
<form
  className="signin-form"
  onSubmit={[Function]}
>
  <h1
    className="mb-3 text-center"
  >
    Sign In
  </h1>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      aria-describedby="emailHelp"
      className="form-control"
      disabled={false}
      id="name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="form-control"
      disabled={false}
      id="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="alert alert-danger"
    role="alert"
    style={
      Object {
        "display": "none",
      }
    }
  >
    Неверный логин или пароль.
  </div>
  <button
    className="btn btn-primary signin-form__button"
    disabled={false}
    type="submit"
  >
    Log In
  </button>
</form>
`;

exports[`Sign in should show an error 1`] = `
<form
  className="signin-form"
  onSubmit={[Function]}
>
  <h1
    className="mb-3 text-center"
  >
    Sign In
  </h1>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <input
      aria-describedby="emailHelp"
      className="form-control"
      disabled={false}
      id="name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="form-control"
      disabled={false}
      id="password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="alert alert-danger"
    role="alert"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Неверный логин или пароль.
  </div>
  <button
    className="btn btn-primary signin-form__button"
    disabled={false}
    type="submit"
  >
    Log In
  </button>
</form>
`;
